garden_ai/model.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum, auto
+from typing import List
+from uuid import uuid4, UUID
+import pydantic
+from pydantic import BaseModel, Field, validator
+
+
+class TempField(Enum):
```

Comment:
This seems complicated... what is going on here that's not just built into pydantic?

Reply:
I wonder if DOI will be set in the client or added in the server anyway. If that's the case then we have our 100% required fields, which are set in the constructor.. everything else is 100% optional, so we don't need any fancy footwork.

Even if the DOI is fetched in the client, then it's not technically a required field. We would just go set it if it's empty and carry on. I think we building extra complexity that is not actually required